pkg/parsers/parsers_test.go

```diff
@@ -26,37 +27,30 @@ func TestParseHost(t *testing.T) {
 		"fd":   "Invalid bind address format: fd",
 	}
 	valids := map[string]string{
-		"0.0.0.1:5555":      "tcp://0.0.0.1:5555",
-		"0.0.0.1:5555/path": "tcp://0.0.0.1:5555/path",
-		":6666":             "tcp://127.0.0.1:6666",
-		":6666/path":        "tcp://127.0.0.1:6666/path",
-		"tcp://:7777":       "tcp://127.0.0.1:7777",
-		"tcp://:7777/path":  "tcp://127.0.0.1:7777/path",
-		// as there's a TrimSpace in there, we should test for it.
+		"0.0.0.1:":                "tcp://0.0.0.1:2376",
+		"0.0.0.1:5555":            "tcp://0.0.0.1:5555",
+		"0.0.0.1:5555/path":       "tcp://0.0.0.1:5555/path",
+		":6666":                   "tcp://127.0.0.1:6666",
+		":6666/path":              "tcp://127.0.0.1:6666/path",
+		"":                        defaultHOST,
+		" ":                       defaultHOST,
```

Comment:
they might feel weird, but they have always been valid - so I thought I'd
document the reality

On Tue, Sep 1, 2015 at 10:12 PM, Doug Davis <notifications@github.com>
wrote:

> In pkg/parsers/parsers_test.go
> <https://github.com/docker/docker/pull/15927#discussion_r38411441>:
>
> > @@ -26,37 +27,30 @@ func TestParseHost(t *testing.T) {
> >  		"fd":   "Invalid bind address format: fd",
> >  	}
> >  	valids := map[string]string{
> > -		"0.0.0.1:5555":      "tcp://0.0.0.1:5555",
> > -		"0.0.0.1:5555/path": "tcp://0.0.0.1:5555/path",
> > -		":6666":             "tcp://127.0.0.1:6666",
> > -		":6666/path":        "tcp://127.0.0.1:6666/path",
> > -		"tcp://:7777":       "tcp://127.0.0.1:7777",
> > -		"tcp://:7777/path":  "tcp://127.0.0.1:7777/path",
> > -		// as there's a TrimSpace in there, we should test for it.
> > +		"0.0.0.1:":                "tcp://0.0.0.1:2376",
> > +		"0.0.0.1:5555":            "tcp://0.0.0.1:5555",
> > +		"0.0.0.1:5555/path":       "tcp://0.0.0.1:5555/path",
> > +		":6666":                   "tcp://127.0.0.1:6666",
> > +		":6666/path":              "tcp://127.0.0.1:6666/path",
> > +		"":                        defaultHOST,
> > +		" ":                       defaultHOST,
>
> don't these (the spaces ones) feel a little weird to treat as valid?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/15927/files#r38411441>.
>